src/federation-info-plugin/components/SchemaFieldsTable/SchemaFieldsTable.jsx

```diff
@@ -0,0 +1,121 @@
+
+import {
+  fieldArgsToValue,
+  fieldTypeToValue,
+  fieldOwnerServicesToValue,
+  fieldReferencedByToValue
+} from '../../utils/schemaFieldToTableCellValue'
+
+const FieldHeader = ({ showReference, order, orderBy, createSortHandler }) => {
+  return (
+    <>
+      <TableCell>
+        <TableSortLabel
+          active={orderBy === 'name'}
+          direction={orderBy === 'name' ? order : 'asc'}
+          onClick={createSortHandler('name')}
+        >
+          Name
+        </TableSortLabel>
+      </TableCell>
+      <TableCell>
+        <TableSortLabel
+          active={orderBy === 'input'}
+          direction={orderBy === 'input' ? order : 'asc'}
+          onClick={createSortHandler('input')}
+        >
+          Input
+        </TableSortLabel>
+      </TableCell>
+      <TableCell>
+        <TableSortLabel
+          active={orderBy === 'type'}
+          direction={orderBy === 'type' ? order : 'asc'}
+          onClick={createSortHandler('type')}
+        >
+          Type
+        </TableSortLabel>
+      </TableCell>
+      <TableCell>
+        <TableSortLabel
+          active={orderBy === 'ownerServices'}
+          direction={orderBy === 'ownerServices' ? order : 'asc'}
+          onClick={createSortHandler('ownerServices')}
+        >
+          Owner service
+        </TableSortLabel>
+      </TableCell>
+      {showReference && (
+        <TableCell>
+          <TableSortLabel
+            active={orderBy === 'referencedBy'}
+            direction={orderBy === 'referencedBy' ? order : 'asc'}
+            onClick={createSortHandler('referencedBy')}
+          >
+            Referenced by
+          </TableSortLabel>
+        </TableCell>
+      )}
+    </>
+  )
+}
+
+const FieldRow = ({ field, showReference }) => (
+  <TableRow key={field.name} hover>
+    <TableCell>{field.name}</TableCell>
+    <TableCell>{fieldArgsToValue(field.args)}</TableCell>
+    <TableCell>{fieldTypeToValue(field.type)}</TableCell>
+    <TableCell>{fieldOwnerServicesToValue(field.ownerServices)}</TableCell>
+    {showReference && (
+      <TableCell>{fieldReferencedByToValue(field.referencedBy)}</TableCell>
+    )}
+  </TableRow>
+)
+
+/**
+ * The component also accepts all the properties defined into the
+ * `SchemaOperationTable` component.
+ *
+ * @param {Function} props.onSortChange called when the user changes the sort order of one of the table's column
+ *
+ * @returns {JSX.Element}
+ */
+const SchemaFieldsTable = ({ onSortChange, ...rest }) => {
+  const [order, setOrder] = useState('desc')
+  const [orderBy, setOrderBy] = useState(null)
+
+  const createSortHandler = property => () => {
+    const isAsc = orderBy === property && order === 'asc'
+    setOrder(isAsc ? 'desc' : 'asc')
+    setOrderBy(property)
+
+    onSortChange && onSortChange(property, order)
+  }
+
+  return (
+    <SchemaOperationTable
+      {...rest}
+      headerRender={({ showReference }) => (
```

Comment:
I think we could get away not passing the `showReference` in this function, but we have it in the props passed to `SchemaFieldsTable` (currently inside the `...rest`).  Also no need for the headerRender to even be a function

Basically something like this:

```jsx
const SchemaFieldsTable = ({ onSortChange, showReference,  ...rest }) => {
  const [order, setOrder] = useState('desc')
  const [orderBy, setOrderBy] = useState(null)

  const createSortHandler = property => () => {
    const isAsc = orderBy === property && order === 'asc'
    setOrder(isAsc ? 'desc' : 'asc')
    setOrderBy(property)

    onSortChange && onSortChange(property, order)
  }

  return (
    <SchemaOperationTable
      {...rest}
      headerRender={
        <FieldHeader
          showReference={showReference}
          order={order}
          orderBy={orderBy}
          createSortHandler={createSortHandler}
        />
      }
      rowRender={({ field }) => (
        <FieldRow
          key={field.name}
          field={field}
          showReference={showReference}
        />
      )}
    />
  )
}
```

We could technically "pre-generate" and memoize rowRenderer, but I don't think is worht the hassle for now

Reply:
Commit pushed based on the solution we discussed on the call.